pymc3/distributions/multivariate.py

```diff
@@ -690,6 +692,145 @@ def logp(self, x):
         )
 
 
+class DirichletMultinomial(Discrete):
+    R"""Dirichlet Multinomial log-likelihood.
+
+    Dirichlet mixture of Multinomials distribution, with a marginalized PMF.
+
+    .. math::
+
+    f(x \mid n, a) = \frac{\Gamma(n + 1)\Gamma(\sum a_k)}
+                              {\Gamma(\n + \sum a_k)}
+                         \prod_{k=1}^K
+                         \frac{\Gamma(x_k +  a_k)}
+                              {\Gamma(x_k + 1)\Gamma(a_k)}
+
+    ==========  ===========================================
+    Support     :math:`x \in \{0, 1, \ldots, n\}` such that
+                :math:`\sum x_i = n`
+    Mean        :math:`n \frac{a_i}{\sum{a_k}}`
+    ==========  ===========================================
+
+    Parameters
+    ----------
+    n : int or array
+        Total counts in each replicate. If n is an array its shape must be (N,)
+        with N = a.shape[0]
+
+    a : one- or two-dimensional array
+        Dirichlet parameter. Elements must be strictly positive.
+        The number of categories is given by the length of the last axis.
+
+    shape : integer tuple
+        Describes shape of distribution. For example if n=array([5, 10]), and
+        p=array([1, 1, 1]), shape should be (2, 3).
+    """
+
+    def __init__(self, n, a, shape, *args, **kwargs):
+
+        super().__init__(shape=shape, defaults=("_defaultval",), *args, **kwargs)
```

Comment:
Dirichlet distribution makes use of `get_test_value` function to compute its [distribution shape](https://github.com/pymc-devs/pymc3/blob/1769258e459e8f40aa8a56e0ac911aa99e7f67de/pymc3/distributions/multivariate.py#L466). Can we use `get_test_value` to determine shape here as well? Doing so, will even us help in #4379. 

Ping @brandonwillard to ask how does `get_test_value` function work?

Reply:
I don't think it is as simple, since the shape can be influenced by the `n` parameter as well as the `a`, whereas in the Dirichlet all information is necessarily contained in the `a` (when shape is not specified)

Reply:
Also that functionality is wrapped in a DeprecationWarning (even though I don't seem to be able to trigger it)

Reply:
@ricardoV94 , just a follow up, it indeed makes sense to avoid the use of `get_test_value` function as also discussed here https://github.com/pymc-devs/pymc3/pull/4000#issuecomment-660307240